frontend/src/components/OutlineButton.vue

```diff
@@ -0,0 +1,106 @@
+<script setup lang="ts">
+import { ref, onMounted, computed } from 'vue';
+
+import ButtonText from './design-system/typography/ButtonText.vue';
+import IconArrowForward from './icons/IconArrowForward.vue';
+import IconArrowBack from './icons/IconArrowBack.vue';
+
+type PropsType = {
+  title: string;
+  type?: 'normal' | 'disabled';
+  icon?: 'left' | 'right' | null;
+  dense?: boolean;
+};
+
+const props = withDefaults(defineProps<PropsType>(), {
+  title: 'Outlined',
+  type: 'normal',
+  icon: null,
+  dense: false,
+});
+
+const buttonType = ref<string>('button');
+
+const buttonHeight = computed(() => {
+  return props.dense ? 'dense' : null;
+});
+
+onMounted(() => {
+  if (props.type === 'disabled') {
+    buttonType.value = 'buttonDisabled';
```

Comment:
Není nutné vytvářet extra třídu na disabled tlačítko, řeš to přes attribut na tlačítku. Ve stylech pak styluješ button s disabled attributem. Pokud to řešíš jen přes styly, tak tlačítko bude jen vypadat, jako disabled, ale ve skutečnosti bude plně funkční.

[Jak řešit disabled tlačítko](https://thewebdev.info/2022/03/14/how-to-disable-a-button-in-vue-js/)